### PR TITLE
Adds a "Copy ID" button to context menu

### PR DESCRIPTION
**Pull request by erxclau**
#### What's this PR do?

texastribune PR: https://github.com/texastribune/texastribune/pull/4652
tt-docker-base PR: https://github.com/texastribune/tt-docker-base/pull/587

- Updates development scripts in `package.json`
   - Previously, we had used `webpack` to compile our TypeScript source code. Webpack was removed at some point but the scripts in `package.json` were not updated. We can use the `tsc` CLI instead to compile our code. I updated the `build` and `watch` command. I should note that I only tested the `watch` command a few times because I didn't really have a use for it.
   - Makes note of these scripts in `README`.
- Updates `package-lock.json` to use [lockfile version](https://docs.npmjs.com/cli/v8/configuring-npm/package-lock-json#lockfileversion) 2.
   - This was a byproduct of me using an `npm` version after v7. Most of our Docker-ized repositories still use version 1 lockfiles. I can downgrade my `npm` version to get a version 1 lockfile if that's preferred. 
- Creates a `createCopyIdButton` function.
   - This function creates a `div` in the context menu that says "Copy ID". It has an `onclick` event listener that will write the ID of the media file to the clipboard.
   - [`navigator.clipboard.writeText`](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/writeText) has good browser support. 
   - However, note that it only works in [HTTPS or localhost](https://stackoverflow.com/a/51823007) environments. When outside of these environments, `navigator.clipboard` is undefined. 
   - The only case the clipboard is undefined is when we are developing on [local.texastribune.org](http://local.texastribune.org:8000). Apparently, [local.texastribune.org](http://local.texastribune.org:8000) is not considered a localhost environment. See the manual testing for information about how to circumvent this.
   - I haven't been able to come up with a good `document.execCommand` fallback for this.
- Changes function name `createCopyButton` to `createCopyUrlButton`.
   - Given that we have a new `createCopyIdButton` function, `createCopyButton` might've been confusing as a name.
- Update `createCopyUrlButton` to use `navigator.clipboard`
   - I have a `document.execCommand` fallback for this.
   - Note that we don't have to select anything for the `document.execCommand` fallback because in order to get to the context menu, we have selected the media file already (and this selection defaults to the correct URL).
   - The production implementation without `navigator.clipboard` works on Chrome and Firefox but does not work on Safari.
   - This `navigator.clipboard` implementation works on Chrome, Firefox, and Safari (in viable environments).

#### Why are we doing this? How does it help us?

This allows the photo team to more easily copy a photo ID.

#### How should this be manually tested?

- *In the `docs` folder*
   1. Clone this repository
   2. Run `npm install`
   3. Run `npm run doc`
   4. Spin up a local server for the `doc` directory (`python -m http.server --directory=docs`)
   5. Go to <http://127.0.0.1:8000/UserInterfaceTutorial.html>
   6. Select the `memory` folder and then right click
   7. Click on the "Copy ID" button
   8. Paste somewhere. The ID should be 3
   9. Try the same with the `local` folder or try uploading your own files.
- *In the `texastribune` app*
   1. Pull the `ttfiles-copy-id` branch
   3. On Chrome, we can go to <chrome://flags> to add <http://local.texastribune.org> to the "Insecure origins treated as secure" flag. Enable the flag too.
       - `navigator.clipboard` doesn't work on <http://local.texastribune.org> without this flag. We can't navigate to localhost instead because we can't authenticate through Auth0 on localhost. We can put this on staging if we want to test it out without having to enable this flag.
   5. Then we can follow similar steps as to testing in the docs folder.

#### How should this change be communicated to end users?

Tell the photo team

#### Are there any smells or added technical debt to note?

No

#### What are the relevant tickets?

<https://airtable.com/appyo1zuQd8f4hBVx/tbloNZu8GkM52NKFR/viwPxWENGdckjxRrP/recporrDL4PmEL3aT?blocks=hide>

#### Have you done the following, if applicable:
***(optional: add explanation between parentheses)***

* [ ] Added automated tests? *( )*
* [ ] Tested manually on mobile? *(I don't think its possible to right click on an iPhone)*
* [ ] Checked BrowserStack? *( )*
* [ ] Checked for performance implications? *( )*
* [ ] Checked accessibility? *( )*
* [ ] Checked for security implications? *( )*
* [ ] Updated the documentation/wiki? *(TODO separately from this PR)*

#### TODOs / next steps:

* [ ] Update docs (I reverted the doc update here so as to not clutter the PR)
* [ ] Bump the version

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,24 +1,46 @@
 {
   "name": "ttfiles",
-  "version": "2.6.0",
-  "lockfileVersion": 1,
+  "version": "2.6.1",
+  "lockfileVersion": 2,
```
Comment (ashley-hebler) [inline package-lock.json:4]: re: your PR note about this - seems fine to me to be on version 2. We compile this locally and I'd imagine most of us are on newer node/npm versions as well anyway.

**Review by ashley-hebler — APPROVED** [review]
This makes sense and LGTM. Thanks for all the extra effort you put into the docs and PR explanation.

I was able to spin up the docs locally and see the copy id feature working. Then I could see the option in local texastribune dev in chrome, but saw that it didn't copy. I'm guessing that's because of the localhost limitations you mentioned. Anyway, this is very cool and impressive that you could jump into this app and make changes like this.

:shipit:

**Comment by ashley-hebler** [comment]
@erxclau -Consider this a take it or leave it suggestion, but I was thinking about the mobile limitation...

Would something like this work?
Screen Shot 2022-04-22 at 10 32 44 AM
Where the currently selected item displays above the table in plain text that they could highlight/copy/paste?

I def. don't think it's a hold up bc it's way outside of the scope of what we proposed to Pu and I'm not even sure if it's a good UI pattern/ a heavy lift. BUT I just wanted to throw that out there while you're versioning all the things. I know with it all broken a part like this, incremental changes are tedious.

**Comment by erxclau** [comment]
> @erxclau -Consider this a take it or leave it suggestion, but I was thinking about the mobile limitation...
> 
> Would something like this work? Screen Shot 2022-04-22 at 10 32 44 AM Where the currently selected item displays above the table in plain text that they could highlight/copy/paste?
> 
> I def. don't think it's a hold up bc it's way outside of the scope of what we proposed to Pu and I'm not even sure if it's a good UI pattern/ a heavy lift. BUT I just wanted to throw that out there while you're versioning all the things. I know with it all broken apart like this, incremental changes are tedious.

@ashley-hebler I can look into something like this for next sprint. I think it could be more involved than the context menu button.

**Pull request merged**